File: src/test/starterkit/starterAI.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

public class Game
{
    public Player[] Players = new Player[2];
    public Table Dishwasher;
    public Table Window;
    public Table Blueberry;
    public Table IceCream;
    public List<Table> Tables = new List<Table>();
}

public class Table
{
    public Position Position;
    public bool HasFunction;
    public Item Item;
}

public class Item
{
    public string Content;
    public bool HasPlate;
    public Item(string content){
        Content = content;
        HasPlate = Content.Contains(MainClass.Dish);
    }
}

public class Player
{
    public Position Position;
    public Item Item;
    public Player(Position position, Item item){
        Position = position;
        Item = item;
    }
    public void Update(Position position, Item item){
        Position = position;
        Item = item;
    }
}

public class Position
{
    public int X, Y;
    public Position(int x, int y){
        X = x;
        Y = y;
    }

    public int Manhattan(Position p2) => Math.Abs(X - p2.X) + Math.Abs(Y - p2.Y);

    public override string ToString()
    {
        return X + " " + Y;
    }
}



public class MainClass
{
    public static bool Debug = true;
    public const string Dish = "DISH";

    public static Game ReadGame(){
        var game = new Game();
        game.Players[0] = new Player(null, null);
        game.Players[1] = new Player(null, null);

        for (int i = 0; i < 7; i++)
        {
            string kitchenLine = ReadLine();
            for (var x = 0; x < kitchenLine.Length; x++){
                if (kitchenLine[x] == 'W') game.Window = new Table { Position = new Position(x, i), HasFunction = true };
                if (kitchenLine[x] == 'D') game.Dishwasher = new Table { Position = new Position(x, i), HasFunction = true };
                if (kitchenLine[x] == 'I') game.IceCream = new Table { Position = new Position(x, i), HasFunction = true };
                if (kitchenLine[x] == 'B') game.Blueberry = new Table { Position = new Position(x, i), HasFunction = true };
                if (kitchenLine[x] == '#') game.Tables.Add(new Table { Position = new Position(x, i) });
            }
        }

        return game;
    }

    private static void Move(Position p) => Console.WriteLine("MOVE " + p);

    private static void Use(Position p){
        Console.WriteLine("USE " + p + "; C# Starter AI");
    }

    private static string ReadLine(){
        var s = Console.ReadLine();
        if (Debug)
            Console.Error.WriteLine(s);
        return s;
    }


    static void Main()
    {
        string[] inputs;

        // ALL CUSTOMERS INPUT: to ignore until Bronze
        int numAllCustomers = int.Parse(ReadLine());
        for (int i = 0; i < numAllCustomers; i++)
        {
            inputs = ReadLine().Split(' ');
            string customerItem = inputs[0]; // the food the customer is waiting for
            int customerAward = int.Parse(inputs[1]); // the number of points awarded for delivering the food
        }

        // KITCHEN INPUT
        var game = ReadGame();

        while (true)
        {
            int turnsRemaining = int.Parse(ReadLine());

            // PLAYERS INPUT
            inputs = ReadLine().Split(' ');
            game.Players[0].Update(new Position(int.Parse(inputs[0]), int.Parse(inputs[1])), new Item(inputs[2]));
            inputs = ReadLine().Split(' ');
            game.Players[1].Update(new Position(int.Parse(inputs[0]), int.Parse(inputs[1])), new Item(inputs[2]));

            //Clean other tables
            foreach(var t in game.Tables){
                t.Item = null;
            }
            int numTablesWithItems = int.Parse(ReadLine()); // the number of tables in the kitchen that currently hold an item
            for (int i = 0; i < numTablesWithItems; i++)
            {
                inputs = ReadLine().Split(' ');
                var table = game.Tables.First(t => t.Position.X == int.Parse(inputs[0]) && t.Position.Y == int.Parse(inputs[1]));
                table.Item = new Item(inputs[2]);
            }

            inputs = ReadLine().Split(' ');
            string ovenContents = inputs[0]; // ignore until bronze league
            int ovenTimer = int.Parse(inputs[1]);
            int numCustomers = int.Parse(ReadLine()); // the number of customers currently waiting for food
            for (int i = 0; i < numCustomers; i++)
            {
                inputs = ReadLine().Split(' ');
                string customerItem = inputs[0];
                int customerAward = int.Parse(inputs[1]);
            }

            // GAME LOGIC
            // fetch a dish, pick ice cream and drop the dish on an empty table
            var myChef = game.Players[0];
            if (!myChef.Item?.HasPlate ?? false)
                Use(game.Dishwasher.Position);
            else if(!myChef.Item.Content.Contains("ICE_CREAM"))
                Use(game.IceCream.Position);
            // once ready, put it on the first empty table for now
            else
                Use(game.Tables.First(t => t.Item == null).Position);
        }
    }
}

File: src/test/starterkit/starterAI.vb
Imports System.Collections.Generic
Module CodeALaMode

    Public Debug As Boolean = True
    Public Const Dish As String = "DISH"

    Public Class Game
        Public Players As Player() = New Player(1) {}
        Public Dishwasher As Table
        Public Window As Table
        Public Blueberry As Table
        Public IceCream As Table
        Public Tables As New List(Of Table)

        Function getFirstEmptyTable() As Table
            For Each t As Table In Tables
                If t.Item Is Nothing Then Return t
            Next
            Throw New InvalidOperationException
        End Function
        Function getTableAtPosition(p As Position) As Table
            For Each t As Table In Tables
                If t.Position.X = p.X AndAlso t.Position.Y = p.Y Then
                    Return t
                End If
            Next
            Throw New InvalidOperationException
        End Function
    End Class

    Public Class Table
        Public Position As Position
        Public HasFunction As Boolean
        Public Item As Item
    End Class

    Public Class Item
        Public Content As String
        Public HasPlate As Boolean
        Public Sub New(content As String)
            Me.Content = content
            Me.HasPlate = content.Contains(CodeALaMode.Dish)
        End Sub
    End Class

    Public Class Player
        Public Position As Position
        Public Item As Item
        Public Sub New(Position As Position, Item As Item)
            Me.Position = Position
            Me.Item = Item
        End Sub
        Public Sub Update(position As Position, item As Item)
            Me.Position = position
            Me.Item = item
        End Sub
    End Class

    Public Class Position
        Public X, Y As Integer
        Public Sub New(x As Integer, y As Integer)
            Me.X = x
            Me.Y = y
        End Sub

        Public Function Manhattan(p2 As Position) As Integer
            Return Math.Abs(X - p2.X) + Math.Abs(Y - p2.Y)
        End Function

        Public Overrides Function ToString() As String
            Return X & " " & Y
        End Function
    End Class

    Public Function ReadGame() As Game
        Dim game As Game = New Game()
        game.Players(0) = New Player(Nothing, Nothing)
        game.Players(1) = New Player(Nothing, Nothing)

        For i As Integer = 0 To 7 - 1
            Dim kitchenLine As String = ReadLine()
            For x As Integer = 0 To kitchenLine.Length - 1
                Dim t As New Table
                Select Case kitchenLine(x)
                    Case "W"c
                        t.Position = New Position(x, i) : t.HasFunction = True
                        game.Window = t
                    Case "D"c
                        t.Position = New Position(x, i) : t.HasFunction = True
                        game.Dishwasher = t
                    Case "I"c
                        t.Position = New Position(x, i) : t.HasFunction = True
                        game.IceCream = T
                    Case "B"c
                        T.Position = New Position(x, i) : T.HasFunction = True
                        game.Blueberry = T
                    Case "#"c
                        t.Position = New Position(x, i)
                        game.Tables.Add(t)
                End Select
            Next
        Next

        Return game
    End Function

    Private Sub Move(p As Position)
        Console.WriteLine("MOVE " & p.ToString())
    End Sub

    Private Sub Use(p As Position)
        Console.WriteLine("USE " & p.ToString() & "; VB Starter AI")
    End Sub

    Private Function ReadLine() As String
        Dim s As String = Console.ReadLine()
        If Debug Then Console.Error.WriteLine(s)
        Return s
    End Function


    Sub Main()
        Dim inputs As String()

        ' ALL CUSTOMERS INPUT to ignore until Bronze
        Dim numAllCustomers As Integer = CInt(ReadLine())
        For i As Integer = 0 To numAllCustomers - 1
            inputs = ReadLine().Split(" "c)
            Dim customerItem As String = inputs(0) ' the food the customer Is waiting for
            Dim customerAward As Integer = CInt(inputs(1)) ' the number Of points awarded For delivering the food
        Next

        ' KITCHEN INPUT
        Dim Game As Game = ReadGame()

        While True
            Dim turnsRemaining As Integer = CInt(ReadLine())

            ' PLAYERS INPUT
            inputs = ReadLine().Split(" "c)
            Game.Players(0).Update(New Position(CInt(inputs(0)), CInt(inputs(1))), New Item(inputs(2)))
            inputs = ReadLine().Split(" "c)
            Game.Players(1).Update(New Position(CInt(inputs(0)), CInt(inputs(1))), New Item(inputs(2)))

            'Clean other tables
            For Each t As Table In Game.Tables
                t.Item = Nothing
            Next
            Dim numTablesWithItems As Integer = CInt(ReadLine()) ' the number Of tables In the kitchen that currently hold an item
            For i As Integer = 0 To numTablesWithItems - 1
                inputs = ReadLine().Split(" "c)
                Dim position As New Position(CInt(inputs(0)), CInt(inputs(1)))
                Dim Table as Table = Game.getTableAtPosition(position)
                Table.Item = New Item(inputs(2))
            Next

            inputs = ReadLine().Split(" "c)
            Dim ovenContents As String = inputs(0) ' ignore until bronze league
            Dim ovenTimer As Integer = CInt(inputs(1))
            Dim numCustomers As Integer = CInt(ReadLine()) ' the number Of customers currently waiting For food
            For i As Integer = 0 To numCustomers - 1
                inputs = ReadLine().Split(" "c)
                Dim customerItem As String = inputs(0)
                Dim customerAward As Integer = CInt(inputs(1))
            Next

            ' GAME LOGIC
            ' fetch a dish, pick ice cream And drop the dish on an empty table
            Dim myChef As Player = Game.Players(0)
            Dim myChefhasPlate As Boolean = If(myChef.Item IsNot Nothing, myChef.Item.HasPlate, False)
            If (Not myChefhasPlate) Then
                Use(Game.Dishwasher.Position)
            ElseIf (Not myChef.Item.Content.Contains("ICE_CREAM")) Then
                Use(Game.IceCream.Position)
                ' once ready, put it on the first empty table for now
            Else
                Use(Game.getFirstEmptyTable.Position)
            End If
        End While
    End Sub

End Module
